js/ui/data_grid.d.ts

```diff
@@ -2685,9 +2684,10 @@ export interface ColumnHeaderFilter {
    * @docid GridBaseColumn.headerFilter.dataSource
    * @type_function_param1_field1 component:object
    * @default undefined
-   * @type_function_return void
+   * @type_function_return Array<any>|Store|DataSourceOptions
+   * @type Array<any>|Store|DataSourceOptions
    */
-  dataSource?: Array<any> | Store | ((options: { component?: any; dataSource?: DataSourceOptions }) => any) | DataSourceOptions;
+  dataSource?: FilterLookupDataSource<any> | (() => FilterLookupDataSource<any>);
```

Comment:
AFAIK, there's no need to return anything from this function.

https://js.devexpress.com/Documentation/21_2/Guide/UI_Components/DataGrid/How_To/Customize_Header_Filter_Data_Source/#Change_the_Generated_Data_Source